driver/SDL/sdl_common.c

```diff
@@ -263,6 +261,6 @@ uint32_t keycode_to_ctrl_key(SDL_Keycode sdl_key)
             return LV_KEY_PREV;
 
         default:
-            return '\0';
+            return sdl_key;
```

Comment:
Is this correct to return `sdl_key` here?  
Until now we only returned `ctrl_key` (or LVGL codes)

Reply:
When there is no way to map the SDL_KEY to a LVGL key we can return the sdl_key. This could be useful if you try to map events to other keys.

Reply:
But how do you guarantee there are no "collisions"?  
Could some `sdl_key` value be the same as another `LV_KEY_*`, so that the result is ambiguous?

Reply:
It does not look like they are conflicting. It seems the constants for the lv_keys were used from sdl:

>>> lv.KEY.__dict__
{'UP': 17, 'DOWN': 18, 'RIGHT': 19, 'LEFT': 20, 'ESC': 27, 'DEL': 127, 'BACKSPACE': 8, 'ENTER': 10, 'NEXT': 9, 'PREV': 11, 'HOME': 2, 'END': 3}

https://wiki.libsdl.org/SDLKeycodeLookup

Reply:
That's interesting.

@kisvegabor - Could you confirm it's OK for an input driver to return to LVGL the SDK key codes directly?

If it is, that means:
- We assume that LVGL key code will always match SDL key codes
- LVGL might receive codes that are not part of the `LV_KEY_*` enum

Reply:
`LV_KEY_*`s are non printable characters and it's fine  to pass normal character, like 'a', 'b', 'c' as they are.  E.g. in [this example](https://docs.lvgl.io/master/examples.html#simple-text-area) you can type anything into the Text area with your keyboard.

Reply:
> `LV_KEY_*`s are non printable characters and it's fine to pass normal character, like 'a', 'b', 'c' as they are. E.g. in [this example](https://docs.lvgl.io/master/examples.html#simple-text-area) you can type anything into the Text area with your keyboard.

@kisvegabor Right. But is it also ok for a `read_cb` callback of an LVGL input device to update `key` to something that is neither a normal character nor `LV_KEY_*`?  
Here we return `sdl_key` so LVGL might be getting some unexpected codes - SDL codes which are not regular characters and are missing (or unmatching with) `LV_KEY_*` enum.

Maybe that's ok - I would just like to make sure it really is.

Reply:
> Maybe that's ok - I would just like to make sure it really is.

Confirmed, it's fine this way.